core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -674,16 +674,18 @@ private boolean checkConsistencyInternal(Inode inode, AlluxioURI path)
       return true;
     }
     if (!inode.isPersisted()) {
-      return !ufs.exists(ufsPath);
+      if (inode.isDirectory()) {
+        return !ufs.isDirectory(ufsPath);
```

Comment:
what if the ufs path is a file? This would return `true` now, but before it would return `false`?

(same as two lines below)